tests/integration/test_specfile.py

```diff
@@ -684,8 +681,35 @@ def test_update_version(
     assert spec.expanded_version == version
 
 
-def test_trailing_newline(spec_autosetup, spec_no_trailing_newline):
-    spec = Specfile(spec_autosetup)
+def test_trailing_newline(specfile_factory, spec_autosetup, spec_no_trailing_newline):
+    spec = specfile_factory(spec_autosetup)
     assert str(spec)[-1] == "\n"
-    spec = Specfile(spec_no_trailing_newline)
+    spec = specfile_factory(spec_no_trailing_newline)
     assert str(spec)[-1] != "\n"
+
+
+def test_reload(specfile_factory, spec_minimal, spec_traditional):
+    spec = specfile_factory(spec_minimal)
+    before_reload = copy.deepcopy(spec)
+
+    if spec.path is not None:
+        os.remove(spec_minimal)
+        shutil.copy2(spec_traditional, spec_minimal)
```

Comment:
I think this should have the same effect and you can drop the extra imports.

```suggestion
        spec_minimal.unlink()
        spec_minimal.write_bytes(spec_traditional.read_bytes())
```